core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -286,6 +286,7 @@ class DAGScheduler(
       case None =>
         // We are going to register ancestor shuffle dependencies
         getAncestorShuffleDependencies(shuffleDep.rdd).foreach { dep =>
+          assert(!shuffleToMapStage.get(dep.shuffleId).isDefined)
```

Comment:
@markhamstra Thank you for your review.
I'll remove the assertion for now but notice that there is a case that some stages refer the overwritten stage as its parent by the overwrite, which is undesirable for `DAGScheduler`.